src/NServiceBus.Core/Sagas/SimpleSaga.cs

```diff
@@ -0,0 +1,85 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Linq.Expressions;
+    using System.Reflection;
+
+    /// <summary>
+    /// This class is used to define sagas containing data and handling a message.
+    /// To handle more message types, implement <see cref="IHandleMessages{T}" />
+    /// for the relevant types.
+    /// To signify that the receipt of a message should start this saga,
+    /// implement <see cref="IAmStartedByMessages{T}" /> for the relevant message type.
+    /// </summary>
+    /// <typeparam name="TSagaData">A type that implements <see cref="IContainSagaData" />.</typeparam>
+    public abstract class SimpleSaga<TSagaData> : Saga where TSagaData : IContainSagaData, new()
+    {
+
+        /// <summary>
+        /// Gets the name of the correlation property for <typeparamref name="TSagaData"/>.
+        /// </summary>
+        protected abstract string CorrelationPropertyName { get; }
```

Comment:
thanks for the explanation. I expected this sort of arguments and I'm good with the current approach, since it's fairly simple code using `nameof`. My main concern with this approach is the discoverability for users, as it's not clear from the naming, that you want the name of the saga data correlation property (although it's the only reasonable option by looking at a full implementation, but that's a longer thought process). Maybe we could rename it to `SagaCorrelationPropertyName` instead?